carbon-design-system/carbon-addons-iot-react
# 2366 MapCard related content panels and react-dnd upgrade

Closes #2366

**Summary**
Upgrades `react-dnd` to the latest version from 11.1.3 to 14.0.2 and the `react-dnd-html5-backend` from 11.1.3 to 14.0.0.
This was needed to use the latest hook based version of the drop target and being able to add it to the MapCard. The MapCard can now accept multiple refs and an example of how to add custom draggable panels was added to storybook and the readme file. 

**Change List (commits, features, bugs, etc)**

1. Upgrade  `react-dnd` and `react-dnd-html5-backend`
2. Updated the useDndProviderElement hook to work with the latest version
3. Added a mergeRefs function to the DragAndDropUtils.js that allows us to use multiple refs for the same component
4. Added a `dropRef` prop to MapCard
5. Created a new demo file called MapboxDragPanelExample.jsx that showcases how to create custom draggable panels
6. Updated tests
7. Updated docs
8. Added MapboxDragPanelExample in a story

![image](https://user-images.githubusercontent.com/5167091/121144901-b54a3900-c83e-11eb-9e7c-6b8cfaa39c33.png)


**Acceptance Test (how to verify the PR)**

- Try out the new story with draggable panels https://deploy-preview-2380--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-☢%EF%B8%8F-mapcard--mapbox-drag-panels-story
- Proof read the new section "Custom draggable panels" in the readme


**Regression Test (how to make sure this PR doesn't break old functionality)**
- Try out all stories using drag and drop. That should be any story with either SimpleList or Table column reordering
